diesel/src/sqlite/connection/serialized_database.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unsafe_code)]
+extern crate libsqlite3_sys as ffi;
+
+/// `SerializedDatabase` is a wrapper for a serialized database that is dynamically allocated by calling `sqlite3_serialize`.
+/// This RAII wrapper is necessary to deallocate the memory when it goes out of scope with `sqlite3_free`.
+#[derive(Debug)]
+pub struct SerializedDatabase {
+    data: *mut u8,
+    len: usize,
+}
+
+impl SerializedDatabase {
+    /// Creates a new `SerializedDatabase` with the given data pointer and length.
+    pub fn new(data: *mut u8, len: usize) -> Self {
+        Self { data, len }
+    }
+
+    /// Returns a slice of the serialized database.
+    pub fn as_slice(&self) -> &[u8] {
```

Comment:
I would like to see an additional comment on this unsafe block that the pointer is never null because we don't pass the `NO_COPY` flag.